packages/typespec-python/generator/pygen/codegen/models/parameter.py

```diff
@@ -232,7 +232,11 @@ def entries(self) -> List["BodyParameter"]:
     def is_form_data(self) -> bool:
         # hacky, but rn in legacy, there is no formdata model type, it's just a dict
         # with all of the entries splatted out
-        return self.type.is_form_data or bool(self.entries)
+        return (
+            self.type.is_form_data
+            or bool(self.entries)
+            or ("multipart/form-data" in self.content_types and self.code_model.options["from_typespec"])
```

Comment:
why do we need to check if it's "from_typespec". We should also remove this option and deal with the differences earlier on in the pipeline

Reply:
Without `from_typespec`, there will be diff for legacy code under `autorest.python` folder. Since multipart is mainly for typespec design, I think we had better keep legacy same as before.